File: documentation-examples-java/src/java/dev/typr/foundations/docs/core/StrategyOverride.java
package dev.typr.foundations.docs.core;

import dev.typr.foundations.QueryListener;
import dev.typr.foundations.Transactor;

@SuppressWarnings("unused")
public class StrategyOverride {
    Transactor tx = null; // placeholder
    QueryListener logger = QueryListener.NOOP; // placeholder
    //start
    Transactor txWithLogging = tx.withStrategy(
        Transactor.defaultStrategy().withListener(logger));
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/core/StrategyOverride.scala
package dev.typr.foundations.docs.core
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*

import dev.typr.foundations.QueryListener

@SuppressWarnings(Array("unused"))
object StrategyOverride:
  val tx: Transactor = null // placeholder
  val logger: QueryListener = QueryListener.NOOP
  //start
  val txWithLogging: Transactor =
    tx.withStrategy(Strategy.empty().withListener(logger))
  //stop
